packages/SwingSet/src/message.js

```diff
@@ -28,4 +28,160 @@ export function insistMessage(message) {
       X`message has non-string non-null .result ${message.result}`,
     );
   }
+  return undefined;
+}
+
+/**
+ * @param {unknown} vdo
+ * @returns { VatDeliveryObject, asserts vdo is VatDeliveryObject }
```

Comment:
Looks like tsc doesn't grok this:

> An asserts return type predicate implies that the returned value is of type void, and there is no provision for returning values of other types. -- https://github.com/microsoft/TypeScript/pull/32695 <- https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions

Reply:
Hm, ok I was wondering if that was legit. I think I was trying to support two simultaneous use patterns:

```js
function receiveFoo(allegedFoo) {
  insistFoo(allegedFoo); // defined with '@returns asserts' but no return value
  /** @type Foo */
  const foo = allegedFoo;
  thingWhichTakesFoo(foo);
}
```

and:

```js
function receiveFoo(allegedFoo) {
  /** @type Foo */
  const foo = insistFoo(allegedFoo); // defined with '@returns Foo' but no asserts
  thingWhichTakesFoo(foo);
}
```

What's the recommended practice for this sort of thing?

One call site for this is in `supervisor-subprocess-xsnap.js`:

https://github.com/Agoric/agoric-sdk/blob/330338daec7e08f1ab7844de2c549b30582dac55/packages/SwingSet/src/kernel/vatManager/supervisor-subprocess-xsnap.js#L221-L234

where we receive data from off-process and need to "convert"/confirm it into a `VatDeliveryObject`.

Reply:
Incidentally, did `tsc` emit an error for this that I missed? I came to that syntax experimentally (and thank you for the docs pointer, I searched for that but failed), but if my experiments are missing input data, I'd like to know where else to look.

Reply:
> Incidentally, did `tsc` emit an error for this that I missed?

I doubt it. I think when `tsc` sees bad syntax in jsdoc, it just figures you didn't mean to write typescript and goes into `any` mode.

Reply:
> What's the recommended practice for this sort of thing?

#72 is not yet closed, but we seem to be leaning toward this one:

> ```js
>   insistFoo(allegedFoo); // defined with '@returns asserts' but no return value
> ```

Reply:
fixed, I think